src/pytorch_lightning/utilities/migration/utils.py

```diff
@@ -37,7 +37,7 @@ def migrate_checkpoint(checkpoint: _CHECKPOINT) -> Tuple[_CHECKPOINT, Dict[str,
         checkpoints and objects that do not support being deep-copied.
     """
     ckpt_version = _get_version(checkpoint)
-    if LooseVersion(ckpt_version) > LooseVersion(pl.__version__):
+    if Version(ckpt_version) > Version(pl.__version__):
```

Comment:
do we know if it works fine with rc|dev|post inversion string?

Reply:
It should. This is what `module_available` uses too